Workflow/Sources/WorkflowAction.swift

```diff
@@ -46,13 +48,30 @@ public struct AnyWorkflowAction<WorkflowType: Workflow>: WorkflowAction {
             return
         }
         self._apply = { return base.apply(toState: &$0) }
+        self.base = base
     }
 
     /// Creates a type-erased workflow action with the given `apply` implementation.
     ///
     /// - Parameter apply: the apply function for the resulting action.
-    public init(_ apply: @escaping (inout WorkflowType.State) -> WorkflowType.Output?) {
-        self._apply = apply
+    public init(
+        _ apply: @escaping (inout WorkflowType.State) -> WorkflowType.Output?,
+        fileID: StaticString = #fileID,
+        line: UInt = #line
+    ) {
+        let closureAction = ClosureAction<WorkflowType>(
```

Comment:
could we expose a way for consumers who receive `AnyWorkflowAction` to know if this is wrapping a closure vs. a real action? could maybe make `ClosureAction` public to achieve that

```
let action: AnyWorkflowAction<SomeWorkflow> = ...

if action.base is ClosureAction<SomeWorkflow> {

}
```

Reply:
yeah we can do that. do you think full access to the underlying type is needed? or would a flag suffice?

Reply:
In general my default is "provide more access" but I know as the SDK author its the opposite for you :p for now I suppose a flag would work